### ISSUE TITLE
NTLK invalid package metdata breaks install via poetry

### ISSUE THREAD

**Issue by chrisegner** [nltk/nltk#2766]
It appears poetry is enforcing PEP440 on package metadata.  The NLTK uses package metadata that does not conform and so poetry fails to install (with poetry 1.2.0 and later, it seems - 1.1.4 doesn't seem affected).

```
[[package]]
name = "nltk"
version = "3.6.2"
description = "Natural Language Toolkit"
category = "main"
optional = false
python-versions = ">=3.5.*"
```

Error message (in docker)
```
STEP 14: RUN poetry install
Creating virtualenv science-lens-BYYjfetP-py3.8 in /home/1001/.cache/pypoetry/virtualenvs
Installing dependencies from lock file

  InvalidVersion

  Invalid PEP 440 version: '3.5.'

  at /usr/local/lib/python3.8/site-packages/poetry/core/version/pep440/parser.py:67 in parse
       63│     @classmethod
       64│     def parse(cls, value: str, version_class: Optional[Type["PEP440Version"]] = None):
       65│         match = cls._regex.search(value) if value else None
       66│         if not match:
    →  67│             raise InvalidVersion(f"Invalid PEP 440 version: '{value}'")
       68│ 
       69│         if version_class is None:
       70│             from poetry.core.version.pep440.version import PEP440Version
       71│ 

The following error occurred when trying to handle this error:


  ValueError

  Could not parse version constraint: >=3.5.*

  at /usr/local/lib/python3.8/site-packages/poetry/core/semver/helpers.py:139 in parse_single_constraint
      135│ 
      136│         try:
      137│             version = Version.parse(version)
      138│         except ValueError:
    → 139│             raise ValueError(
      140│                 "Could not parse version constraint: {}".format(constraint)
      141│             )
      142│ 
      143│         if op == "<":
subprocess exited with status 1
subprocess exited with status 1
error building at STEP "RUN poetry install": exit status 1
level=error msg="exit status 1"
```

**Comment by tomaarsen** [comment]
@chrisegner 

As you know, poetry will add NLTK version `3.6.2` to it's poetry.lock file, with 
```
[[package]]
name = "nltk"
version = "3.6.2"
description = "Natural Language Toolkit"
category = "main"
optional = false
python-versions = ">=3.5.*"
```
This version [3.6.2](https://github.com/nltk/nltk/releases/tag/3.6.2) was released on April 20th 2021.
Afterwards, the `">=3.5.*"` was replaced with just `">=3.5"` in #2720 on May 29th 2021.
(Since then, just yesterday, `">=3.5"` was replaced with `">=3.6"`, but that's besides the point)

Removing `.*` as was done back in May has solved this issue, and made NLTK compliant with [PEP440](https://www.python.org/dev/peps/pep-0440/) again.

However, no official release was drafted since that patch. This has caused poetry to take the latest version - i.e. from April, which does not have this fix. Perhaps @stevenbird can draft a new release soon?

<details>
<summary>Until this is done, you can resolve your issue this way. (Click here to expand)</summary>

Follow the output of `poetry add --help`:

```
DESCRIPTION
  The add command adds required packages to your pyproject.toml and installs them.

  If you do not specify a version constraint, poetry will choose a suitable one based on the available package versions.

  You can specify a package in the following forms:
    - A single name (requests)
    - A name and a constraint (requests@^2.23.0)
    - A git url (git+https://github.com/python-poetry/poetry.git)
    - A git url with a revision (git+https://github.com/python-poetry/poetry.git#develop)
    - A file path (../my-package/my-package.whl)
    - A directory (../my-package/)
    - A url (https://example.com/packages/my-package-0.1.0.tar.gz)
```

In particular, `A git url with a revision (git+https://github.com/python-poetry/poetry.git#develop)` might work. 

i.e.
```
poetry add git+https://github.com/nltk/nltk.git#develop
```

Doing this updated my poetry.lock to:
```
[[package]]
name = "nltk"
version = "3.6.2"
description = "Natural Language Toolkit"
category = "main"
optional = false
python-versions = ">=3.6"
develop = false
```

Which should allow correct installations.
Naturally this isn't a particularly permanent solution, as you would rather just use `poetry add nltk`, but it will (hopefully) work until a new `nltk` release is created.

**Disclaimer: I've never used poetry myself**
</details>

- Tom Aarsen

**Comment by nomicode** [comment]
I did a little digging. the commit that introduced the bug is here: https://github.com/nltk/nltk/commit/b047b0da0e0cc5b22017ea2e4c0e0a8018bd3798

I used the commit date (Sep 25, 2020) to find the latest release on https://github.com/nltk/nltk/releases that was _before_ this commit. the release in question is https://github.com/nltk/nltk/releases/tag/3.5

so, until a new version of nltk is release that includes the fix for this issue, one workaround that doesn't involve tinkering with poetry's auto-generated files is to pin the version to 3.5:

```
poetry add nltk@3.5
```

I did this, and the problem seems to be fixed

**Comment by tomaarsen** [comment]
This problem should be resolved since NLTK 3.6.3. Feel free to let us know if the problem persists.
